src/blocktree_processor.rs

```diff
@@ -207,37 +206,48 @@ pub fn process_blocktree(
             entry_height += entries.len() as u64;
         }
 
-        match meta.next_slots.len() {
-            0 => {
-                let next_blob_index = {
-                    if meta.is_full() {
-                        0
-                    } else {
-                        meta.consumed
-                    }
-                };
-                // Reached the end of this fork.  Record the final entry height and last entry id
-                bank_forks_info.push(BankForksInfo {
-                    bank_id,
-                    entry_height,
-                    last_entry_id,
-                    next_blob_index,
-                })
-            }
-            1 => pending_slots.push((meta.next_slots[0], bank_id, entry_height, last_entry_id)),
-            _ => {
-                // This is a fork point, create a new child bank for each fork
-                pending_slots.extend(meta.next_slots.iter().map(|next_slot| {
-                    let child_bank = Bank::new_from_parent(&bank);
-                    trace!("Add child bank for slot={}", next_slot);
-                    let child_bank_id = *next_slot;
-                    bank_forks.insert(child_bank_id, child_bank);
-                    (*next_slot, child_bank_id, entry_height, last_entry_id)
-                }));
-            }
+        let slot_complete = leader_scheduler
+            .read()
+            .unwrap()
+            .num_ticks_left_in_slot(bank.tick_height())
+            == 0;
+
+        if !slot_complete || meta.next_slots.is_empty() {
+            // Reached the end of this fork.  Record the final entry height and last entry id
+
+            bank_forks_info.push(BankForksInfo {
+                bank_id: slot,
+                entry_height,
+                last_entry_id,
+                next_blob_index: meta.consumed,
```

Comment:
@carllin this is what I'd like to see, but it's failing the 2 node 1 tick/slot test...